### PR TITLE
Make ClusterRoles deployment configurable

### PR DESCRIPTION
**Pull request by Allex1**
Addresses https://github.com/open-telemetry/opentelemetry-helm-charts/issues/239

### CODE REVIEW

File: charts/opentelemetry-operator/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: opentelemetry-operator
-version: 0.8.2
+version: 0.8.4
```
Comment (TylerHelmuth) [inline charts/opentelemetry-operator/Chart.yaml:3]:
```suggestion
version: 0.8.3
```

File: charts/opentelemetry-operator/templates/clusterrole.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+{{- if .Values.rbac.create }}
```
Comment (TylerHelmuth) [inline charts/opentelemetry-operator/templates/clusterrole.yaml:1]: There are multiple ClusterRoles defined in this file.  Is there value in being able to enabled/disable each cluster role individually?  Or does it not ever make sense with the operator to have 1 cluster role but not the others.

File: charts/opentelemetry-operator/templates/clusterrolebinding.yaml [hunk 3]
```diff
@@ -1,3 +1,4 @@
+{{- if .Values.rbac.create }}
```
Comment (TylerHelmuth) [inline charts/opentelemetry-operator/templates/clusterrolebinding.yaml:1]: Same questions here

File: charts/opentelemetry-operator/values.yaml [hunk 4]
```diff
@@ -62,6 +62,11 @@ admissionWebhooks:
       # kind:
       # name:
 
+## Create the provided ClusterRoles and ClusterRoleBindings
+##
+rbac:
+  create: true
```
Comment (TylerHelmuth) [inline charts/opentelemetry-operator/values.yaml:68]:
```suggestion
  enabled: true
```
The charts usually use `enabled` as the field to turn things on and off

File: charts/opentelemetry-operator/templates/clusterrole.yaml (hunk 2 above)
Comment (Allex1) [reply charts/opentelemetry-operator/templates/clusterrole.yaml:1]: I would have the flag to switch between all or nothing. The issue in a multi tenant k8s cluster is that you cannot deploy ClusterRole at all.

File: charts/opentelemetry-operator/templates/clusterrolebinding.yaml (hunk 3 above)
Comment (Allex1) [reply charts/opentelemetry-operator/templates/clusterrolebinding.yaml:1]: I would have the flag to switch between all or nothing. The issue in a multi tenant k8s cluster is that you cannot deploy ClusterRole at all.

**Review by TylerHelmuth — APPROVED** [review]
@dmitryax

File: charts/opentelemetry-operator/values.yaml [hunk 5]
```diff
@@ -62,6 +62,11 @@ admissionWebhooks:
       # kind:
       # name:
 
+## Create the provided ClusterRoles and ClusterRoleBindings
+##
+rbac:
+  enabled: true
```
Comment (dmitryax) [inline charts/opentelemetry-operator/values.yaml:68]: I think we should use the same interface as we have in the collector chart:

```
clusterRole:
  create: true
```

Comment (dmitryax) [reply charts/opentelemetry-operator/values.yaml:68]: Unless you you think that that interface is not the best (we will need to change collector chart in that case then)

Comment (Allex1) [reply charts/opentelemetry-operator/values.yaml]: @dmitryax changed it to your suggestion

**Review by dmitryax — APPROVED** [review]

**Comment by Allex1** [comment]
@dmitryax @TylerHelmuth fixed conflicts,  pls restart CI

**Review by dmitryax — APPROVED** [review]

**Pull request merged**